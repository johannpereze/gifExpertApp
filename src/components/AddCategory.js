import { useState } from "react";
export const AddCategory = () => {
  const [inputValue, setInputValue] = useState("Hola Mundo");

  const handleInputChange = ()=>{
      
  }

  return (
    <>
      <input type="text" value={inputValue} onChange={console.log} />
    </>
  );
};
