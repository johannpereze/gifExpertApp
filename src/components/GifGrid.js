// import { useEffect, useState } from "react";
// import { getGifs } from "../helpers/getGifs";

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";
import PropTypes from "prop-types"
require("dotenv").config();

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className={"animate__animated animate__fadeIn"}>{category}</h3>
      {loading && <p>Cargando...</p>}
      <div className={"card-grid"}>
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};


GifGrid.prototypes ={
  category: PropTypes.string.isRequired
}