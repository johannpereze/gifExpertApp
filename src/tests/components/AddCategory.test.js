import { AddCategory } from "../../components/AddCategory"
import { shallow } from 'enzyme'

describe('Pruebas del componente AddCategory', () => {

    const setCategories = jest.fn()

    let wrapper = shallow(<AddCategory setCategories={setCategories} />); //Lo dejo duplicado para no perder el tipado

    beforeEach(() => {
        jest.clearAllMocks() //realmente no cambia nada en este caso
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    })

    test('Debería devolver el snapshot correctamente', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('Debería cambiar la caja de texto', () => {

        const input = wrapper.find('input');

        const value = 'Hola Mundo';

        input.simulate('change', {
            target: {
                value: value
            }
        })

        const inputAfter = wrapper.find("input");
        expect(inputAfter.prop("value")).toBe(value);

    })

    test('No debe de postear la información con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } })

        expect(setCategories).not.toHaveBeenCalled()
    })

    test('debe de llamar el setCategories y limpiar la caja de texto', () => {

        //1. simular inputchange
        const value = 'Hola Mundo';
        const inputToChange = wrapper.find('input')
        inputToChange.simulate('change', {
            target: {
                value: value
            }
        })

        const inputBefore = wrapper.find("input");
        expect(inputBefore.prop("value")).toBe(value);

        //2. Simular el submit
        const submit = wrapper.find('form');
        submit.simulate('submit', {
            preventDefault: jest.fn()
        })

        expect(setCategories).toBeCalled()

        const inputAftere = wrapper.find("input");
        expect(inputAftere.prop("value")).toBe('');

    })


})


