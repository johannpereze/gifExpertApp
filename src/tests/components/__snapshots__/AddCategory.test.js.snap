// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AddCategory Deber√≠a devolver el snapshot correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
